12-watchtower-API.md

```diff
@@ -270,73 +317,58 @@ Sample code (python) for the client to prepare the `encrypted_blob`:
 
 **Micropayments**. A small payment is sent to the WatchTower for every new job (i.e. over the lightning network)
 
-**Subscription**. WatchTower is periodically rewarded / paid for their service to the customer. (i.e. over the lightning network or dirty fiat subscription). 
+**Subscription**. WatchTower is periodically rewarded / paid for their service to the customer. (i.e. over the lightning network or fiat subscription). 
 
 Both micropayments and subscriptions are favourable for a WatchTower.
 
-**Problems with onchain bounty** We highlight that the on-chain bounty approach is not ideal for a watching network. It lets the customer hire N WatchTowers (O(N) storage for each tower) and only one WatchTower will be rewarded upon collecting the bounty. If this approach was sought, then the miners may end up becoming WatchTowers as they can very easily front-run an entire watching network. Thus to reduce the power and responsibility of miners, we should avoid this approach and advocate that miners do not participate in a watching network. 
- 
-## Number of updates
-
-To offer full protection, a WatchTower requires an encrypted blob for every single update in the channel. This is a symptom of replace-by-revocation channels as there is a single valid state and a set of "revoked" states. Only one of the states can be broadcast and this is up to the counterparty (cheater) to decide. 
+**Problems with onchain bounty** We highlight that the on-chain bounty approach is not ideal for a watching network. It lets the customer hire N WatchTowers (O(N) storage for each tower) and only one WatchTower will be rewarded upon collecting the bounty. On top of that, the onchain bounty allows a network-wise DoS attack for free.
 
-All signed receipts only correspond to a single job / justice transaction. Thus the customer should keep a copy of all signed receipts received from the WatchTower. A future BOLT can extend the signed receipt to include a committment to all previous encrypted blobs (merkle tree) to reduce this storage. 
 
 ## No compression of justice transaction 
 
-There are tricks using [hashchains](https://github.com/rustyrussell/ccan/blob/master/ccan/crypto/shachain/design.txt) to reduce the storage requirements for each justice transaction. For this BOLT, we have decided to keep the hiring protocol simpe in order to get WatchTowers up and running. Stroage is relatively cheap and we can revisit this standard if it becomes a problem. 
+The storage requirements for a WatchTower can be reduced (linearly) by implementing [shachain](https://github.com/rustyrussell/ccan/blob/master/ccan/crypto/shachain/design.txt), therefore storing the parts required to build the transaction and the corresponding signing key instead of the full transaction. For this BOLT, we have decided to keep the hiring protocol simple. Storage is relatively cheap and we can revisit this standard if it becomes a problem. 
```

Comment:
maybe when it becomes a problem. it is defo going to explode. i wonder how hard it is to add the compression. i.e. we could offer a discount for customers who do that.

Reply:
This is part of your writing, feel free to change it / remove it if you like

Reply:
i think its important to have compression. this is more of a thought than a comment. is there ways to mitigate that. i.e. if we end up storing 100 mb for a customer, we could just ask them to close / re-open it, and we'll create their account with the difference.

Reply:
I can add it as a todo/fixme for future iterations. I'm not sure what's the community opinion towards sharing the one-time keys with the Watchers. IRC it was designed in that way in part to allow it, but I feel opinions have changed.

Reply:
oh you mean the output to steal the coins, the key can be given to the watch tower? that would be pretty cool, but then you really have to trust the watch tower. i dont think we want to have that custody

Reply:
That's basically what the shachain approach is. It's basically the same key that you give to the counterpart once the old state has been revoked. Each side of the channel has already exchanged this info so they will also give it to the tower.

Reply:
ah yes ok i remember.